crates/core/src/cache.rs

```diff
@@ -207,13 +208,23 @@ impl TryFrom<&Addon> for AddonCacheEntry {
             let mut folder_names: Vec<_> = addon.folders.iter().map(|a| a.id.clone()).collect();
             folder_names.sort();
 
+            let external_release_id = if repository == RepositoryKind::TownlongYak {
+                addon.file_id().map(ExternalReleaseId::FileId)
+            } else {
+                addon
+                    .version()
+                    .map(str::to_string)
+                    .map(ExternalReleaseId::Version)
+            };
+
```

Comment:
This is how we cache the version that was just installed / updated. When we reload the addon from the cache, it'll use this as the version / file_id to check against the remote API. This resolves issues where the addon .toc version is not comparable to the remote version on the API